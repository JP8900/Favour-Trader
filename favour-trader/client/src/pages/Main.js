import React, {Component} from 'react';
import ContractModal from '../components/ContractModal.js';
import axios from 'axios'
import {Row} from 'reactstrap';
import MatchCard from "../components/MatchCard";

class Main extends Component {
    constructor() {
        super();
        this.state = {
            matchedUsers: [],
            selectedUser: {},
            modalOpen: false,
        };
        this.toggleModal = this.toggleModal.bind(this);
        this.renderMatches = this.renderMatches.bind(this);
    }

    componentDidMount() {
        const { authService } = this.props;
        if (authService.loggedIn()) {
            const config = {
                headers: {
                    Authorization: authService.getToken()
                }
            };

            axios.get('/api/users/matches', config)
            .then(res => res.data.matches)
            .then(matches => this.setState({matchedUsers: matches }))
            .catch( (err) => {
                console.log(err);
            });
        }
        // else should redirect to login page
    }

    toggleModal(user) {
        this.setState({
            selectedUser: user,
            modalOpen: !this.state.modalOpen,
        });
    }

    renderMatches(matches) {
        return (
            <div>
                {
                    (matches) ? (
                        <div className={'container'}>
                            <Row>
                                {
                                    matches.map(function (match) {
                                        return (<MatchCard key={match._id} user={match}
                                                           reveal={this.toggleModal.bind(this, match)}/>)
                                    }, this)
                                }
                            </Row>
                        </div>
                    ) : (
                        ''
                    )
                }
            </div>
        );
    }

    render() {
        return (
            <div>
                <header className={'d-flex justify-content-center'}>
                    <h1 className={'center-helper'}>Welcome to FavourTrader</h1>
                </header>
                <div className={'center-helper'}>
                    {this.renderMatches(this.state.matchedUsers)}
                </div>
                <ContractModal isOpen={this.state.modalOpen} toggle={this.toggleModal.bind(this, {})}
                               user={this.state.selectedUser}/>
            </div>
        );
    }
}

export default Main;