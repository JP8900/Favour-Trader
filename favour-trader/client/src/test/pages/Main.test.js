import React from 'react';
import renderer from 'react-test-renderer';
import Main from '../../pages/Main.js';
import AuthServiceMock from '../components/AuthServiceMock.js';

it('renders correctly', () => {
  var authServiceMock = new AuthServiceMock();
  const tree = renderer
    .create(<Main authService={authServiceMock}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});