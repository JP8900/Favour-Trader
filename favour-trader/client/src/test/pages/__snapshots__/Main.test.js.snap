// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    className="d-flex justify-content-center"
  >
    <h1
      className="center-helper"
    >
      Welcome to FavourTrader
    </h1>
  </header>
  <div
    className="center-helper"
  >
    <div>
      <div
        className="container"
      >
        <div
          className="row"
        />
      </div>
    </div>
  </div>
  <div>
    
  </div>
</div>
`;
